openmdao/utils/general_utils.py

```diff
@@ -222,7 +222,10 @@ def ensure_compatible(name, value, shape=None, indices=None):
                                  (name, shape, value.shape))
 
     # finally make sure shape of indices is compatible
-    if indices is not None and shape != ind_shape[:len(shape)]:
+    if isinstance(indices, np.ndarray):
+        contains_slice = _is_slice(indices)
+
+    if indices is not None and shape != ind_shape[:len(shape)] and not contains_slice:
```

Comment:
If indices is not None but also not an ndarray, it looks like contains_slice will be undefined.  If the test suite isn't catching this, we probably need another test.